### PR TITLE
Feature/net6 workaround

### PR DESCRIPTION
**Pull request by TristanBoland**
It seems the QRCode Class is no longer available in QRCoder for .NET6.
Throws an "Could not load type 'QRCoder.QRCode' from assembly 'QRCoder, Version=1.4.3.0" exception currently.
I needed this library in one of my projects. 
So i decided to publish my workaround to here. Hopefully it is of use to some

### CODE REVIEW

**Comment by flytzen** [comment]
> @flytzen thoughts? I've been seeing more of these requests, not sure how we want to proceed.

Well, if this works - which it looks like it does - then we should pull it in for now. We do need to test it on Linux, though. Even if it is "windows only", we can still do it as a short-term thing. I'll update the unit tests to run on net6 as well as run the test suite on Linux as well.
But if https://github.com/codebude/QRCoder/issues/315 isn't resolved soon, then I guess we will have to switch to a different QR coder library. 

Relates to #100 and #98 and #94.
If we do switch to another lib then we probably want/need to drop legacy support (see #89) and only support netstandard2.0 up and bump the main version.

@TristanBoland Thank you!!

**Comment by flytzen** [comment]
@ahwm The tests are all passing on Linux as well, so I think we should merge this. I do want to just check a couple of things before we merge :)

File: Google.Authenticator/TwoFactorAuthenticator.cs [hunk 1]
```diff
@@ -81,15 +81,23 @@ private static string GenerateQrCodeUrl(int qrPixelsPerModule, string provisionU
...
+                    using (var qrGenerator = new QRCodeGenerator())
+                    using (var qrCodeData = qrGenerator.CreateQrCode(provisionUrl, QRCodeGenerator.ECCLevel.Q))
+                    using (var qrCode = new QRCode(qrCodeData))
+                    using (var qrCodeImage = qrCode.GetGraphic(qrPixelsPerModule))
+                    using (var ms = new MemoryStream())
+                    {
+                        qrCodeImage.Save(ms, System.Drawing.Imaging.ImageFormat.Png);
+                        qrCodeUrl = $"data:image/png;base64,{Convert.ToBase64String(ms.ToArray())}";
+                    }
+                #endif
```
Comment (flytzen) [inline Google.Authenticator/TwoFactorAuthenticator.cs:100]:
```suggestion
                               using (var qrGenerator = new QRCodeGenerator())
                using (var qrCodeData = qrGenerator.CreateQrCode(provisionUrl, QRCodeGenerator.ECCLevel.Q))
                #if NET6_0
                    using (var qrCode = new PngByteQRCode(qrCodeData))
                    {
                        var qrCodeImage = qrCode.GetGraphic(qrPixelsPerModule);
                        qrCodeUrl = $"data:image/png;base64,{Convert.ToBase64String(qrCodeImage)}";
                    }
                #else
                    using (var qrCode = new QRCode(qrCodeData))
                    using (var qrCodeImage = qrCode.GetGraphic(qrPixelsPerModule))
                    using (var ms = new MemoryStream())
                    {
                        qrCodeImage.Save(ms, System.Drawing.Imaging.ImageFormat.Png);
                        qrCodeUrl = $"data:image/png;base64,{Convert.ToBase64String(ms.ToArray())}";
                    }
                #endif

```

**Review by flytzen — CHANGES_REQUESTED** [review]
This is really great, thank you.
Minor suggestion to avoid duplication.
Slightly more important about adding the `using` statements to avoid resource leaking in high volume scenarios :)

**Comment by TristanBoland** [comment]
I accepted your changes, glad i could help! Was intended as a quick fix so i could go on with my Task, but glad it was of use

**Review by flytzen — APPROVED** [review]
@ahwm I'm approving this. 
I'll do a separate PR with a new version number and a minor fix for a new warning in .Net 6.0.

**Pull request merged**